### ISSUE TITLE
[Adjust Rule] SIM113 Use enumerate only when it's unconditionally executed

### ISSUE THREAD

**Issue by tsx** [MartinThoma/flake8-simplify#54]
## Desired change

* **Rule(s)**: SIM113
* **Adjustment**: Only suggest enumerate when `+=1` is executed unconditionally.

## Explanation

If you pre-filter the things you iterate on before adding 1 to the total, enumerate will have a different number than current total.

## Example

This is an example where the mentioned rule(s) would currently be suboptimal:

```python
odd_x = 0
for x in [1, 2, 3]:
    if x % 2:
        continue
    odd_x += 1
```

**Comment by akx** [comment]
I'd say this is a bug in the rule since if one blindly just applies the change, they'll get a program that behaves differently than before.

**Comment by MartinThoma** [comment]
You are completely right, this is a false-positive. SIM113 triggers way too often.

However, this issue is a duplicate of https://github.com/MartinThoma/flake8-simplify/issues/32 - sadly, I can't see a simple way to fix this issue. Maybe if there is any `continue` it should not trigger?
